File: javascript/7_kyu/halving_sum.js
// Halving Sum
// https://www.codewars.com/kata/5a58d46cfd56cb4e8600009d/train/javascript

function halvingSum(n) {
    var score = 0;
    
    while (n >= 1) {
      score += n;
      
      n = Math.floor((n * 1) / 2);
    }
    
    return score;
  }
  

File: python/7_kyu/halving_sum.py
# Halving Sum
# https://www.codewars.com/kata/5a58d46cfd56cb4e8600009d/train/python

import math

def halving_sum(n): 
    score = 0

    while (n >= 1):
        score += n

        n = math.floor((n * 1) / 2)
    
    return score